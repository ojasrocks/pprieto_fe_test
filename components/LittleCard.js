import Image from 'next/image'
import Link from 'next/link'
import { useEffect, useState } from 'react'
import _ from 'lodash'


export default function LittleCard({value, fav}) {

    const notAvailable = 'https://upload.wikimedia.org/wikipedia/commons/6/64/Poster_not_available.jpg'
    const [heart,setHeart] = useState(fav);
    const [overlay,setOverlay] = useState(false);
    const direct = '/assets/layout/Exports/Icons/icon-heart-'
    const mediaMatch = window.matchMedia('(max-width: 1000px)');
    const [matches, setMatches] = useState(mediaMatch.matches);
  
    useEffect(() => {
      const handler = e => setMatches(e.matches);
      mediaMatch.addListener(handler);
      return () => mediaMatch.removeListener(handler);
    });

    useEffect( ()=>{
        let favourites = JSON.parse(localStorage.getItem('favourites'))
        if (_.isEmpty(favourites)) favourites = [];
        if (heart && !favourites.some(el => el === value['imdbID'])){
            favourites.push(value['imdbID'])
            localStorage.setItem('favourites', JSON.stringify(favourites))
        }
        if (!heart && favourites.some(el => el === value['imdbID'])){
            favourites = favourites.filter(elem => elem !== value['imdbID'])
            localStorage.setItem('favourites', JSON.stringify(favourites))
        }
    },[heart])

    return (
        <div 
        id={value.imdbID}
        style={{
            position: 'relative',
            backgroundImage: `url(${value["Poster"] === "N/A" ? notAvailable : value["Poster"]})`,
            backgroundRepeat: 'no-repeat',
            backgroundSize: matches ? '180px 240px' : '100% 100%',
            maxHeight: '100%',
            maxWidth:'100%',
            backgroundPosition: 'center',
            borderRadius: '4px',
        }}
        onMouseEnter={()=>setOverlay(true)}
        onMouseLeave={()=>setOverlay(false)}
        >
            <div 
            id={`${value.imdbID}_heart`}
            style={{
                position:'absolute',
                top: '12px',
                right: '12px',
                zIndex: 1,
                visibility: heart || overlay ? 'visible' : 'hidden'
            }}><Image 
            src={`${direct}${heart ? 'full' : 'white'}.svg`}
            width={'16px'} 
            height={'16px'} 
            style={{
                display:'absolute',
                backgroundColor:'transparent',
            }}
            onClick={e => {
                e.preventDefault()
                setHeart(!heart)
            }}
            />
            </div>
            <div 
            id={`${value.imdbID}_overlay`}
            style={{
                top: 0,
                left: 0,
                backgroundColor: '#1B2329',
                overflow: 'hidden',
                width: '100%',
                height: '100%',
                opacity: overlay ? 0.9 : 0,
                borderRadius: '4px',
                display:'flex',
                flexFlow: 'column',
                justifyContent:'flex-end'
            }}>
            <Link href={`/posts/${value.imdbID}?fav=${heart}`}>
            <div 
            id={`${value.imdbID}_toclick`}
            style={{
                position:'absolute',
                bottom: 0,
                left: 0,
                right: 0,
                overflow: 'hidden',
                width: '100%',
                maxHeight:'100%',
                padding:'12px',
            }}>
                <div style={{
                    visibility: overlay ? 'visible' : 'hidden',
                    display:'flex',
                    flexFlow:'column',
                    overflowWrap: 'break-word',
                    maxHeight:'100%',
                    justifyContent:'space-evenly',
                }}><h3 style={{
                    color:'white',
                    textOverflow:'ellipsis',
                    marginBottom: 0,
                    fontSize: matches ? '1em' : '1.17em',
                    overflowY:'hidden'
                }}
                >{value["Title"]}</h3
                ><p style={{
                    color:'white',
                    marginTop: matches ? '6px':'12px',
                    marginBottom: 0,
                }}>{value["Year"]}</p></div>
            </div>
            </Link>
            </div>
        </div>
    )
}