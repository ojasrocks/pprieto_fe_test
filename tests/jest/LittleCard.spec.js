import renderer from 'react-test-renderer';
import LittleCard from '../../components/LittleCard';

it('changes the opacity of overlay when hovered', () => {
  const myMockFav = jest.fn();
  const myMockValue = jest.fn();

  myMockFav.mockReturnValueOnce(true).mockReturnValueOnce(false)
  myMockValue.mockReturnValueOnce({
    "Title": "The Trip to Italy",
    "Year": "2014",
    "imdbID": "tt2967006",
    "Type": "movie",
    "Poster": "https://m.media-amazon.com/images/M/MV5BNTAxMDQzNzA3Nl5BMl5BanBnXkFtZTgwNzU4MDk4MTE@._V1_SX300.jpg"
}).mockReturnValueOnce({
    "Title": "My Voyage to Italy",
    "Year": "1999",
    "imdbID": "tt0173772",
    "Type": "movie",
    "Poster": "https://m.media-amazon.com/images/M/MV5BMDRkM2RiZWYtODYxNy00ZmJlLTkyZDQtNDdiNTIwMzkxYzQ1XkEyXkFqcGdeQXVyMzIwNDY4NDI@._V1_SX300.jpg"
})
  for (let i = 0; i < 2; i++){
    const component = renderer.create(
    <LittleCard fav={myMockFav()} value={myMockValue()}/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  renderer.act(() => {
    tree.props.onMouseEnter();
  });
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  renderer.act(() => {
    tree.props.onMouseLeave();
  });
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
}
});