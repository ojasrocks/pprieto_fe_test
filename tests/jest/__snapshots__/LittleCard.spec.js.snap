// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the opacity of overlay when hovered 1`] = `
<div
  id="tt2967006"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "backgroundImage": "url(https://m.media-amazon.com/images/M/MV5BNTAxMDQzNzA3Nl5BMl5BanBnXkFtZTgwNzU4MDk4MTE@._V1_SX300.jpg)",
      "backgroundPosition": "center",
      "backgroundRepeat": "no-repeat",
      "backgroundSize": "180px 240px",
      "borderRadius": "4px",
      "maxHeight": "240px",
      "position": "relative",
    }
  }
>
  <div
    id="tt2967006_heart"
    style={
      Object {
        "position": "absolute",
        "right": "12px",
        "top": "12px",
        "visibility": "visible",
        "zIndex": 1,
      }
    }
  >
    <span
      style={
        Object {
          "background": "none",
          "border": 0,
          "boxSizing": "border-box",
          "display": "inline-block",
          "height": "initial",
          "margin": 0,
          "maxWidth": "100%",
          "opacity": 1,
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "width": "initial",
        }
      }
    >
      <span
        style={
          Object {
            "background": "none",
            "border": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "padding": 0,
            "width": "initial",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2716%27%20height=%2716%27/%3e"
          style={
            Object {
              "background": "none",
              "border": 0,
              "display": "block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "padding": 0,
              "width": "initial",
            }
          }
        />
      </span>
      <img
        data-nimg="intrinsic"
        decoding="async"
        onClick={[Function]}
        onError={[Function]}
        onLoad={[Function]}
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "backgroundColor": "transparent",
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
      <noscript>
        <img
          data-nimg="intrinsic"
          decoding="async"
          loading="lazy"
          onClick={[Function]}
          src="/assets/layout/Exports/Icons/icon-heart-full.svg"
          srcSet="/assets/layout/Exports/Icons/icon-heart-full.svg 1x, /assets/layout/Exports/Icons/icon-heart-full.svg 2x"
          style={
            Object {
              "backgroundColor": "transparent",
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
    </span>
  </div>
  <div
    id="tt2967006_overlay"
    style={
      Object {
        "backgroundColor": "#1B2329",
        "borderRadius": "4px",
        "height": "100%",
        "left": 0,
        "opacity": 0,
        "overflow": "hidden",
        "top": 0,
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "padding": "12px",
          "position": "absolute",
          "right": 0,
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "visibility": "hidden",
          }
        }
      >
        <h3
          style={
            Object {
              "color": "white",
              "marginBottom": 0,
            }
          }
        >
          The Trip to Italy
        </h3>
        <p
          style={
            Object {
              "color": "white",
              "marginBottom": 0,
              "marginTop": "12px",
            }
          }
        >
          2014
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`changes the opacity of overlay when hovered 2`] = `
<div
  id="tt2967006"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "backgroundImage": "url(https://m.media-amazon.com/images/M/MV5BNTAxMDQzNzA3Nl5BMl5BanBnXkFtZTgwNzU4MDk4MTE@._V1_SX300.jpg)",
      "backgroundPosition": "center",
      "backgroundRepeat": "no-repeat",
      "backgroundSize": "180px 240px",
      "borderRadius": "4px",
      "maxHeight": "240px",
      "position": "relative",
    }
  }
>
  <div
    id="tt2967006_heart"
    style={
      Object {
        "position": "absolute",
        "right": "12px",
        "top": "12px",
        "visibility": "visible",
        "zIndex": 1,
      }
    }
  >
    <span
      style={
        Object {
          "background": "none",
          "border": 0,
          "boxSizing": "border-box",
          "display": "inline-block",
          "height": "initial",
          "margin": 0,
          "maxWidth": "100%",
          "opacity": 1,
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "width": "initial",
        }
      }
    >
      <span
        style={
          Object {
            "background": "none",
            "border": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "padding": 0,
            "width": "initial",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2716%27%20height=%2716%27/%3e"
          style={
            Object {
              "background": "none",
              "border": 0,
              "display": "block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "padding": 0,
              "width": "initial",
            }
          }
        />
      </span>
      <img
        data-nimg="intrinsic"
        decoding="async"
        onClick={[Function]}
        onError={[Function]}
        onLoad={[Function]}
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "backgroundColor": "transparent",
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
      <noscript>
        <img
          data-nimg="intrinsic"
          decoding="async"
          loading="lazy"
          onClick={[Function]}
          src="/assets/layout/Exports/Icons/icon-heart-full.svg"
          srcSet="/assets/layout/Exports/Icons/icon-heart-full.svg 1x, /assets/layout/Exports/Icons/icon-heart-full.svg 2x"
          style={
            Object {
              "backgroundColor": "transparent",
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
    </span>
  </div>
  <div
    id="tt2967006_overlay"
    style={
      Object {
        "backgroundColor": "#1B2329",
        "borderRadius": "4px",
        "height": "100%",
        "left": 0,
        "opacity": 0.9,
        "overflow": "hidden",
        "top": 0,
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "padding": "12px",
          "position": "absolute",
          "right": 0,
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "visibility": "visible",
          }
        }
      >
        <h3
          style={
            Object {
              "color": "white",
              "marginBottom": 0,
            }
          }
        >
          The Trip to Italy
        </h3>
        <p
          style={
            Object {
              "color": "white",
              "marginBottom": 0,
              "marginTop": "12px",
            }
          }
        >
          2014
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`changes the opacity of overlay when hovered 3`] = `
<div
  id="tt2967006"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "backgroundImage": "url(https://m.media-amazon.com/images/M/MV5BNTAxMDQzNzA3Nl5BMl5BanBnXkFtZTgwNzU4MDk4MTE@._V1_SX300.jpg)",
      "backgroundPosition": "center",
      "backgroundRepeat": "no-repeat",
      "backgroundSize": "180px 240px",
      "borderRadius": "4px",
      "maxHeight": "240px",
      "position": "relative",
    }
  }
>
  <div
    id="tt2967006_heart"
    style={
      Object {
        "position": "absolute",
        "right": "12px",
        "top": "12px",
        "visibility": "visible",
        "zIndex": 1,
      }
    }
  >
    <span
      style={
        Object {
          "background": "none",
          "border": 0,
          "boxSizing": "border-box",
          "display": "inline-block",
          "height": "initial",
          "margin": 0,
          "maxWidth": "100%",
          "opacity": 1,
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "width": "initial",
        }
      }
    >
      <span
        style={
          Object {
            "background": "none",
            "border": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "padding": 0,
            "width": "initial",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2716%27%20height=%2716%27/%3e"
          style={
            Object {
              "background": "none",
              "border": 0,
              "display": "block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "padding": 0,
              "width": "initial",
            }
          }
        />
      </span>
      <img
        data-nimg="intrinsic"
        decoding="async"
        onClick={[Function]}
        onError={[Function]}
        onLoad={[Function]}
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "backgroundColor": "transparent",
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
      <noscript>
        <img
          data-nimg="intrinsic"
          decoding="async"
          loading="lazy"
          onClick={[Function]}
          src="/assets/layout/Exports/Icons/icon-heart-full.svg"
          srcSet="/assets/layout/Exports/Icons/icon-heart-full.svg 1x, /assets/layout/Exports/Icons/icon-heart-full.svg 2x"
          style={
            Object {
              "backgroundColor": "transparent",
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
    </span>
  </div>
  <div
    id="tt2967006_overlay"
    style={
      Object {
        "backgroundColor": "#1B2329",
        "borderRadius": "4px",
        "height": "100%",
        "left": 0,
        "opacity": 0,
        "overflow": "hidden",
        "top": 0,
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "padding": "12px",
          "position": "absolute",
          "right": 0,
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "visibility": "hidden",
          }
        }
      >
        <h3
          style={
            Object {
              "color": "white",
              "marginBottom": 0,
            }
          }
        >
          The Trip to Italy
        </h3>
        <p
          style={
            Object {
              "color": "white",
              "marginBottom": 0,
              "marginTop": "12px",
            }
          }
        >
          2014
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`changes the opacity of overlay when hovered 4`] = `
<div
  id="tt0173772"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "backgroundImage": "url(https://m.media-amazon.com/images/M/MV5BMDRkM2RiZWYtODYxNy00ZmJlLTkyZDQtNDdiNTIwMzkxYzQ1XkEyXkFqcGdeQXVyMzIwNDY4NDI@._V1_SX300.jpg)",
      "backgroundPosition": "center",
      "backgroundRepeat": "no-repeat",
      "backgroundSize": "180px 240px",
      "borderRadius": "4px",
      "maxHeight": "240px",
      "position": "relative",
    }
  }
>
  <div
    id="tt0173772_heart"
    style={
      Object {
        "position": "absolute",
        "right": "12px",
        "top": "12px",
        "visibility": "hidden",
        "zIndex": 1,
      }
    }
  >
    <span
      style={
        Object {
          "background": "none",
          "border": 0,
          "boxSizing": "border-box",
          "display": "inline-block",
          "height": "initial",
          "margin": 0,
          "maxWidth": "100%",
          "opacity": 1,
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "width": "initial",
        }
      }
    >
      <span
        style={
          Object {
            "background": "none",
            "border": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "padding": 0,
            "width": "initial",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2716%27%20height=%2716%27/%3e"
          style={
            Object {
              "background": "none",
              "border": 0,
              "display": "block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "padding": 0,
              "width": "initial",
            }
          }
        />
      </span>
      <img
        data-nimg="intrinsic"
        decoding="async"
        onClick={[Function]}
        onError={[Function]}
        onLoad={[Function]}
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "backgroundColor": "transparent",
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
      <noscript>
        <img
          data-nimg="intrinsic"
          decoding="async"
          loading="lazy"
          onClick={[Function]}
          src="/assets/layout/Exports/Icons/icon-heart-white.svg"
          srcSet="/assets/layout/Exports/Icons/icon-heart-white.svg 1x, /assets/layout/Exports/Icons/icon-heart-white.svg 2x"
          style={
            Object {
              "backgroundColor": "transparent",
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
    </span>
  </div>
  <div
    id="tt0173772_overlay"
    style={
      Object {
        "backgroundColor": "#1B2329",
        "borderRadius": "4px",
        "height": "100%",
        "left": 0,
        "opacity": 0,
        "overflow": "hidden",
        "top": 0,
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "padding": "12px",
          "position": "absolute",
          "right": 0,
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "visibility": "hidden",
          }
        }
      >
        <h3
          style={
            Object {
              "color": "white",
              "marginBottom": 0,
            }
          }
        >
          My Voyage to Italy
        </h3>
        <p
          style={
            Object {
              "color": "white",
              "marginBottom": 0,
              "marginTop": "12px",
            }
          }
        >
          1999
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`changes the opacity of overlay when hovered 5`] = `
<div
  id="tt0173772"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "backgroundImage": "url(https://m.media-amazon.com/images/M/MV5BMDRkM2RiZWYtODYxNy00ZmJlLTkyZDQtNDdiNTIwMzkxYzQ1XkEyXkFqcGdeQXVyMzIwNDY4NDI@._V1_SX300.jpg)",
      "backgroundPosition": "center",
      "backgroundRepeat": "no-repeat",
      "backgroundSize": "180px 240px",
      "borderRadius": "4px",
      "maxHeight": "240px",
      "position": "relative",
    }
  }
>
  <div
    id="tt0173772_heart"
    style={
      Object {
        "position": "absolute",
        "right": "12px",
        "top": "12px",
        "visibility": "visible",
        "zIndex": 1,
      }
    }
  >
    <span
      style={
        Object {
          "background": "none",
          "border": 0,
          "boxSizing": "border-box",
          "display": "inline-block",
          "height": "initial",
          "margin": 0,
          "maxWidth": "100%",
          "opacity": 1,
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "width": "initial",
        }
      }
    >
      <span
        style={
          Object {
            "background": "none",
            "border": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "padding": 0,
            "width": "initial",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2716%27%20height=%2716%27/%3e"
          style={
            Object {
              "background": "none",
              "border": 0,
              "display": "block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "padding": 0,
              "width": "initial",
            }
          }
        />
      </span>
      <img
        data-nimg="intrinsic"
        decoding="async"
        onClick={[Function]}
        onError={[Function]}
        onLoad={[Function]}
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "backgroundColor": "transparent",
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
      <noscript>
        <img
          data-nimg="intrinsic"
          decoding="async"
          loading="lazy"
          onClick={[Function]}
          src="/assets/layout/Exports/Icons/icon-heart-white.svg"
          srcSet="/assets/layout/Exports/Icons/icon-heart-white.svg 1x, /assets/layout/Exports/Icons/icon-heart-white.svg 2x"
          style={
            Object {
              "backgroundColor": "transparent",
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
    </span>
  </div>
  <div
    id="tt0173772_overlay"
    style={
      Object {
        "backgroundColor": "#1B2329",
        "borderRadius": "4px",
        "height": "100%",
        "left": 0,
        "opacity": 0.9,
        "overflow": "hidden",
        "top": 0,
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "padding": "12px",
          "position": "absolute",
          "right": 0,
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "visibility": "visible",
          }
        }
      >
        <h3
          style={
            Object {
              "color": "white",
              "marginBottom": 0,
            }
          }
        >
          My Voyage to Italy
        </h3>
        <p
          style={
            Object {
              "color": "white",
              "marginBottom": 0,
              "marginTop": "12px",
            }
          }
        >
          1999
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`changes the opacity of overlay when hovered 6`] = `
<div
  id="tt0173772"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "backgroundImage": "url(https://m.media-amazon.com/images/M/MV5BMDRkM2RiZWYtODYxNy00ZmJlLTkyZDQtNDdiNTIwMzkxYzQ1XkEyXkFqcGdeQXVyMzIwNDY4NDI@._V1_SX300.jpg)",
      "backgroundPosition": "center",
      "backgroundRepeat": "no-repeat",
      "backgroundSize": "180px 240px",
      "borderRadius": "4px",
      "maxHeight": "240px",
      "position": "relative",
    }
  }
>
  <div
    id="tt0173772_heart"
    style={
      Object {
        "position": "absolute",
        "right": "12px",
        "top": "12px",
        "visibility": "hidden",
        "zIndex": 1,
      }
    }
  >
    <span
      style={
        Object {
          "background": "none",
          "border": 0,
          "boxSizing": "border-box",
          "display": "inline-block",
          "height": "initial",
          "margin": 0,
          "maxWidth": "100%",
          "opacity": 1,
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "width": "initial",
        }
      }
    >
      <span
        style={
          Object {
            "background": "none",
            "border": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "padding": 0,
            "width": "initial",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2716%27%20height=%2716%27/%3e"
          style={
            Object {
              "background": "none",
              "border": 0,
              "display": "block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "padding": 0,
              "width": "initial",
            }
          }
        />
      </span>
      <img
        data-nimg="intrinsic"
        decoding="async"
        onClick={[Function]}
        onError={[Function]}
        onLoad={[Function]}
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "backgroundColor": "transparent",
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
      <noscript>
        <img
          data-nimg="intrinsic"
          decoding="async"
          loading="lazy"
          onClick={[Function]}
          src="/assets/layout/Exports/Icons/icon-heart-white.svg"
          srcSet="/assets/layout/Exports/Icons/icon-heart-white.svg 1x, /assets/layout/Exports/Icons/icon-heart-white.svg 2x"
          style={
            Object {
              "backgroundColor": "transparent",
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
    </span>
  </div>
  <div
    id="tt0173772_overlay"
    style={
      Object {
        "backgroundColor": "#1B2329",
        "borderRadius": "4px",
        "height": "100%",
        "left": 0,
        "opacity": 0,
        "overflow": "hidden",
        "top": 0,
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "padding": "12px",
          "position": "absolute",
          "right": 0,
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "visibility": "hidden",
          }
        }
      >
        <h3
          style={
            Object {
              "color": "white",
              "marginBottom": 0,
            }
          }
        >
          My Voyage to Italy
        </h3>
        <p
          style={
            Object {
              "color": "white",
              "marginBottom": 0,
              "marginTop": "12px",
            }
          }
        >
          1999
        </p>
      </div>
    </div>
  </div>
</div>
`;
